test/rpc_test.py

```diff
@@ -363,7 +362,6 @@ def test_invalid_names(self):
         # would add extra overhead to the call, and normal use cases won't
         # create a progress group and exit without doing anything. Hence, it is
         # not worthy to introduce the overhead just for this test case.
-        dist.barrier()
```

Comment:
Any reason this was removed? @mrshenli added it in https://github.com/pytorch/pytorch/pull/28376, to fix this test's flakiness.

Reply:
@rohan-varma Since the one successful `dist.init_process_group(..)` on line 302 was removed too, there is no need to wait for all workers to be synced.

Reply:
as RPC is creating a PG internally, it seems we need to call barrier on that internal PG? @xush6528 @rohan-varma

Reply:
@mrshenli We discussed offline,
since "Connection closed by peer" error should not be thrown from `ProcessGroupAgent` constructor,  the problem might be the `destroy_process_group()` call in the python `except Exception:` section in , `process_group_init_backend_handler(..)`, which is called by `_init_rpc_backend(...)`.